### ISSUE TITLE
[LETTERS][a11y-defect-2]: Update address success must announce itself to screen readers and manage keyboard focus

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#111594]
### Point of contact

@1copenut

### Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

### Details

While auditing the Your VA Letters and Documents, I noticed our success alert does not manage focus or announce itself to screen readers like our "Confirm your address" alert does. These should be the same experience so we can ensure screen reader users understand the current state of their address and changes of context. Screenshots attached below.

---
![Image](https://github.com/user-attachments/assets/a979d360-6a3f-4ee7-ab76-f8964696769b)

---
![Image](https://github.com/user-attachments/assets/721c0afe-df7e-4c05-b3fd-7cb8173b6269)

### Reproduction steps

1. Log into Staging with a test user. I've been testing with user 54.
2. Navigate to the letters app, turn on the screen reader of your choosing
3. Click "Edit your address" and add a nonsense value like "#55" to the Street Address Line 2 field
4. Save your address edit
5. Confirm the green success alert appears on screen but doesn't announce itself

### Proposed solution
We should do two things to match what we're doing on the yellow confirmation alert:

1. Set focus on the heading inside the success alert
2. ~~Wrap the success alert in a `<div role="alert">` to match the confirmation alert~~ UPDATE: too noisy for SR

### Acceptance criteria
- [ ] Success alert manages focus correctly on render
- [ ] Success alert announces itself to screen readers correctly on render:
    * NVDA + Chrome
    * NVDA + Firefox
    * Narrator + Edge
    * VoiceOver + Safari
    * (ideally) iOS VoiceOver + mobile Safari

### References, articles, or WCAG support

1. [Understanding SC 1.3.1: Info and Relationships Level A)](https://www.w3.org/WAI/WCAG22/Understanding/info-and-relationships.html)
1. [Understanding SC 1.3.2: Meaningful Sequence (Level A)](https://www.w3.org/WAI/WCAG22/Understanding/meaningful-sequence.html)

### Type of issue

- [ ] Axe-core
- [x] Screenreader | assistive tech & device support
- [x] Keyboard
- [ ] Voice command
- [ ] Zoom
- [ ] Color, typography, & visual elements
- [ ] Components and pattern standards
- [ ] Content organization
- [ ] Focus
- [ ] Headings
- [ ] Content style guide
- [ ] Link and button labels
- [ ] Semantic HTML
- [ ] Markup and metadata
- [ ] Non-text content and media
- [ ] Overall user experience
- [ ] Cognitive | user flows & navigating
- [ ] URL standards
- [ ] Something else

**Comment by dysbo** [comment]
I am only able to test VoiceOver on MacOS, but that was successful ✅
